### PR TITLE
Added support for Application credentials to authenticate openstack.

### LINKED ISSUE #575: [Feature] Implement support for Openstack Application Credentials
**Feature (What you would like to be added):**
[Application Credentials](https://docs.openstack.org/keystone/latest/user/application_credentials.html) can be used instead of username/password to authenticate against Openstack. Backup of etcd to Swift should be possible using application credentials.
**Motivation (Why is this needed?):**
Application credentials are useful when implementing secret rotation scenarios for Openstack.
According to [documentation](https://github.com/gardener/etcd-backup-restore/blob/master/example/storage-provider-secrets/02-openstack-swift-storage-secret.yaml) only username/password is currently allowed.

### PR DESCRIPTION
**Pull request by ishan16696**
**What this PR does / why we need it**:
This PR adds support for passing [Application credentials](https://docs.openstack.org/keystone/latest/user/application_credentials.html) to authenticate Openstack.
This PR supports the backward compatibility, it doesn't removes the old way(username/password) of passing Openstack credentials.

**Which issue(s) this PR fixes**:
Fixes #575 

**Special notes for your reviewer**:

**Release note**:
```noteworthy operator
Added support for Application credentials to authenticate Openstack client for Openstack backup buckets.
```

### CODE REVIEW

**Comment by ishan16696** [comment]
I have tested it out for single node etcd as well as multi-node etcd cluster that etcd cluster is not having a downtime with dynamically updating the backup-storage secrets from [username/password](https://github.com/gardener/etcd-backup-restore/blob/master/example/storage-provider-secrets/02-openstack-swift-storage-secret.yaml#L1-L13) form of credentials to [Application](https://docs.openstack.org/keystone/latest/user/application_credentials.html) form of credentials to authenticate openstack client.

File: pkg/snapstore/swift_snapstore.go [hunk 1]
```diff
@@ -181,50 +233,74 @@ func readSwiftCredentialFiles(dirname string) (*clientconfig.ClientOpts, error)
...
-func readSwiftCredentialDir(dirname string) (*swiftCredentials, error) {
+func readSwiftCredentialDir(dirName string) (*swiftCredentials, error) {
 	cred := &swiftCredentials{}
-	files, err := os.ReadDir(dirname)
+	files, err := os.ReadDir(dirName)
 	if err != nil {
 		return nil, err
 	}
 
+	if cred.AuthType, err = getSwiftCredentialAuthType(files); err != nil {
```
Comment (aaronfern) [inline pkg/snapstore/swift_snapstore.go:243]: Is there a case where we have Username/Password file and ApplicationCredentialID/ApplicationCredentialName/ApplicationCredentialSecret files are both present? 
If that is the case then the flow might be a bit indeterministic

**Review by aaronfern — COMMENTED** [review]
Thanks for the well written PR @ishan16696!
Looks good, just one question

File: pkg/snapstore/swift_snapstore.go (hunk 1 above)
Comment (ishan16696) [reply pkg/snapstore/swift_snapstore.go:243]: >  Username/Password file and ApplicationCredentialID/ApplicationCredentialName/ApplicationCredentialSecret files are both present?

I'm assuming that shouldn't be the case as I have also mentioned in the [doc examples of secrets](https://github.com/gardener/etcd-backup-restore/pull/580/files#diff-ee65c41e7448f5d9a9cb70e797bf558bb21d2bf277348daf329cf8e6261084eaR39-R52).

Comment (unmarshall) [reply pkg/snapstore/swift_snapstore.go:243]: @aaronfern is right. You should not assume and have a validation which prevents this from happening.

**Review by aaronfern — APPROVED** [review]
Thanks for the changes @ishan16696!
/lgtm

**Pull request merged**